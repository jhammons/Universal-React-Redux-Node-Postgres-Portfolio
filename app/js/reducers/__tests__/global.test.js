import globalReducer from '../global';
import {
    TOGGLE_HAMBURGER,
    OPEN_HAMBURGER,
    CLOSE_HAMBURGER,
    SET_HEADER,
    ANIMATE_ABOUT,
    ANIMATE_PORTFOLIO,
    ANIMATE_CONTACT,
    ANIMATE_OFF
} from '../../actions/global';


describe('global reducer', () => {
    const initialState = {
        headerState: false,
        hamburgerState: false,
        animateAbout: false,
        animatePortfolio: false,
        animateContact: false,
        animateOff: false
    };

    test('initial state is correct', () => {
        const action = { type: 'test_action' };

        expect(globalReducer(undefined, action)).toEqual(initialState);
        expect(globalReducer(undefined, action)).toMatchSnapshot();
    });

    test('TOGGLE_HAMBURGER', () => {
        const action = { type: TOGGLE_HAMBURGER };
        const expectedState = {
            ...initialState,
            hamburgerState: true
        };

        expect(globalReducer(undefined, action)).toEqual(expectedState);
        expect(globalReducer(undefined, action)).toMatchSnapshot();
    });

    test('OPEN_HAMBURGER', () => {
        const action = { type: OPEN_HAMBURGER };
        const expectedState = {
            ...initialState,
            hamburgerState: true,
            htmlClass: 'menu-open'
        };

        expect(globalReducer(undefined, action)).toEqual(expectedState);
        expect(globalReducer(undefined, action)).toMatchSnapshot();
    });

    test('CLOSE_HAMBURGER', () => {
        const action = { type: CLOSE_HAMBURGER };
        const expectedState = {
            ...initialState,
            hamburgerState: false,
            htmlClass: null
        };

        expect(globalReducer(undefined, action)).toEqual(expectedState);
        expect(globalReducer(undefined, action)).toMatchSnapshot();
    });

    test('SET_HEADER', () => {
        const action = { type: SET_HEADER, bool: true };
        const expectedState = {
            ...initialState,
            headerState: action.bool
        };

        expect(globalReducer(undefined, action)).toEqual(expectedState);
        expect(globalReducer(undefined, action)).toMatchSnapshot();
    });

    test('ANIMATE_ABOUT', () => {
        const action = { type: ANIMATE_ABOUT, bool: true };
        const expectedState = {
            ...initialState,
            animateAbout: action.bool,
        };

        expect(globalReducer(undefined, action)).toEqual(expectedState);
        expect(globalReducer(undefined, action)).toMatchSnapshot();
    });

    test('ANIMATE_PORTFOLIO', () => {
        const action = { type: ANIMATE_PORTFOLIO, bool: true };
        const expectedState = {
            ...initialState,
            animatePortfolio: true
        };

        expect(globalReducer(undefined, action)).toEqual(expectedState);
        expect(globalReducer(undefined, action)).toMatchSnapshot();
    });

    test('ANIMATE_CONTACT', () => {
        const action = { type: ANIMATE_CONTACT, bool: true };
        const expectedState = {
            ...initialState,
            animateContact: true
        };

        expect(globalReducer(undefined, action)).toEqual(expectedState);
        expect(globalReducer(undefined, action)).toMatchSnapshot();
    });

    test('ANIMATE_OFF', () => {
        const action = { type: ANIMATE_OFF, bool: true };
        const expectedState = {
            ...initialState,
            animateOff: true
        };

        expect(globalReducer(undefined, action)).toEqual(expectedState);
        expect(globalReducer(undefined, action)).toMatchSnapshot();
    });
});