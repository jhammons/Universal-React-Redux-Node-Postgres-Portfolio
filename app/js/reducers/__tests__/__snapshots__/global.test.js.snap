// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global reducer ANIMATE_ABOUT 1`] = `
Object {
  "animateAbout": true,
  "animateContact": false,
  "animateOff": false,
  "animatePortfolio": false,
  "hamburgerState": false,
  "headerState": false,
}
`;

exports[`global reducer ANIMATE_CONTACT 1`] = `
Object {
  "animateAbout": false,
  "animateContact": true,
  "animateOff": false,
  "animatePortfolio": false,
  "hamburgerState": false,
  "headerState": false,
}
`;

exports[`global reducer ANIMATE_OFF 1`] = `
Object {
  "animateAbout": false,
  "animateContact": false,
  "animateOff": true,
  "animatePortfolio": false,
  "hamburgerState": false,
  "headerState": false,
}
`;

exports[`global reducer ANIMATE_PORTFOLIO 1`] = `
Object {
  "animateAbout": false,
  "animateContact": false,
  "animateOff": false,
  "animatePortfolio": true,
  "hamburgerState": false,
  "headerState": false,
}
`;

exports[`global reducer CLOSE_HAMBURGER 1`] = `
Object {
  "animateAbout": false,
  "animateContact": false,
  "animateOff": false,
  "animatePortfolio": false,
  "hamburgerState": false,
  "headerState": false,
  "htmlClass": null,
}
`;

exports[`global reducer OPEN_HAMBURGER 1`] = `
Object {
  "animateAbout": false,
  "animateContact": false,
  "animateOff": false,
  "animatePortfolio": false,
  "hamburgerState": true,
  "headerState": false,
  "htmlClass": "menu-open",
}
`;

exports[`global reducer SET_HEADER 1`] = `
Object {
  "animateAbout": false,
  "animateContact": false,
  "animateOff": false,
  "animatePortfolio": false,
  "hamburgerState": false,
  "headerState": true,
}
`;

exports[`global reducer TOGGLE_HAMBURGER 1`] = `
Object {
  "animateAbout": false,
  "animateContact": false,
  "animateOff": false,
  "animatePortfolio": false,
  "hamburgerState": true,
  "headerState": false,
}
`;

exports[`global reducer initial state is correct 1`] = `
Object {
  "animateAbout": false,
  "animateContact": false,
  "animateOff": false,
  "animatePortfolio": false,
  "hamburgerState": false,
  "headerState": false,
}
`;
